commit 968e6bdfe7b5a3f3f201ec7dadbb2bd108e6a54c (refs/changes/15/3915/2)
Author: Joshua M. Clulow <jmc@joyent.com>
Date:   2018-05-07T19:10:21+00:00 (1 year, 5 months ago)
    
    MORAY-470 moray test suite not running all tests
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>
    Approved by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/Makefile b/Makefile
index 6f7ed05..b914ef9 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@
 #
 
 #
-# Copyright (c) 2016, Joyent, Inc. All rights reserved.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 #
@@ -19,33 +19,51 @@
 #
 # Tools
 #
-NPM		 = npm
-FAUCET		 = ./node_modules/.bin/faucet
-CONFIGURE	 = ./tools/configure
+NPM =				npm
+FAUCET =			./node_modules/.bin/faucet
+CONFIGURE =			./tools/configure
 
 #
 # We use ctrun(1) to ensure that child processes created by the test cases are
 # always cleaned up.  However, on systems that don't provide ctrun(1), this
 # could be commented out.
 #
-CTRUN		 ?= ctrun -o noorphan
+CTRUN ?=			ctrun -o noorphan
 
 
 #
 # Files and other definitions
 #
-JSON_FILES	 = package.json \
-		   etc/moray-test-suite-stock.json \
-		   etc/moray-test-suite-custom-both.json
-JS_FILES	:= tools/configure $(shell find lib test -name '*.js')
-JSL_FILES_NODE	 = $(JS_FILES)
-JSSTYLE_FILES	 = $(JS_FILES)
-JSSTYLE_FLAGS    = -f ./tools/jsstyle.conf
-JSL_CONF_NODE	 = tools/jsl.node.conf
+JSON_FILES =			package.json \
+				etc/moray-test-suite-stock.json \
+				etc/moray-test-suite-custom-both.json
+JS_FILES :=			tools/configure \
+				$(shell find lib test -name '*.js')
+JSL_FILES_NODE =		$(JS_FILES)
+JSSTYLE_FILES =			$(JS_FILES)
+JSSTYLE_FLAGS =			-f ./tools/jsstyle.conf
+JSL_CONF_NODE =			tools/jsl.node.conf
 
-MORAY_TEST_CONFIG_FILE	?= etc/moray-test-suite.json
-MORAY_TEST_RUNDIR        = run
-MORAY_TEST_ENV_FILE	 = $(MORAY_TEST_RUNDIR)/env.sh
+MORAY_TEST_CONFIG_FILE ?=	etc/moray-test-suite.json
+MORAY_TEST_RUNDIR =		run
+MORAY_TEST_ENV_FILE =		$(MORAY_TEST_RUNDIR)/env.sh
+
+#
+# Build a list of "test_<name>" targets from the contents of the "test"
+# directory.  Files that match the pattern "test/<name>.test.js" will be
+# executed automatically by "make test", and any individual test can be invoked
+# via "make test_<name>".
+#
+TESTS =				$(addprefix test_,\
+				    $(subst .test.js$,,\
+				    $(notdir $(wildcard test/*.test.js))))
+
+#
+# Most of the test programs produce TAP output, but at least one produces
+# regular bunyan output under some conditions.
+#
+POST_PROCESS =			$(FAUCET)
+test_loop :			POST_PROCESS = bunyan -lfatal
 
 #
 # Targets
@@ -57,27 +75,14 @@ all:
 CLEAN_FILES += node_modules
 
 .PHONY: test
-test: | $(FAUCET) $(MORAY_TEST_ENV_FILE)
-	(set -o pipefail; \
-	source $(MORAY_TEST_ENV_FILE) && \
-	$(CTRUN) node test/cli-parsing.test.js | $(FAUCET) && \
-	$(CTRUN) node test/cli-sanity.test.js | $(FAUCET) && \
-	$(CTRUN) node test/client.test.js | $(FAUCET) && \
-	$(CTRUN) node test/close-handles.test.js | $(FAUCET) && \
-	$(CTRUN) node test/buckets.test.js | $(FAUCET) && \
-	$(CTRUN) node test/objects.test.js | $(FAUCET) && \
-	$(CTRUN) node test/invalid-buckets.test.js | $(FAUCET) && \
-	$(CTRUN) node test/invalid-fast.test.js | $(FAUCET) && \
-	$(CTRUN) node test/sql.test.js | $(FAUCET) && \
-	$(CTRUN) node test/integ.test.js | $(FAUCET) && \
-	$(CTRUN) node test/arrays.test.js | $(FAUCET) && \
-	$(CTRUN) node test/version.test.js | $(FAUCET) && \
-	$(CTRUN) node test/clientparams.test.js | $(FAUCET) && \
-	$(CTRUN) node test/findobjects.test.js | $(FAUCET) && \
-	$(CTRUN) node test/findobjects-requireindexes.test.js | $(FAUCET) && \
-	$(CTRUN) node test/loop.test.js | bunyan -lfatal )
+test: $(TESTS)
 	@echo tests passed
 
+test_%: test/%.test.js | $(FAUCET) $(MORAY_TEST_ENV_FILE)
+	set -o pipefail && source $(MORAY_TEST_ENV_FILE) && \
+	    $(CTRUN) node $< | $(POST_PROCESS)
+	@echo test $< passed
+
 $(FAUCET): all
 
 $(MORAY_TEST_ENV_FILE): $(MORAY_TEST_CONFIG_FILE)
